src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -141,9 +141,12 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		// events we care about are fired when the document focus is on something else - typically the root window.
 		this._register(DOM.addDisposableListener(window, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
 			// Make sure that the current active element is an ancestor - this is to prevent us from handling events when the focus is
-			// on some other dialog or part of the app. 
+			// on some other dialog or part of the app.
 			if (DOM.isAncestor(this.container.nativeElement, document.activeElement) && this.isActive() && this.model.activeCell) {
 				const event = new StandardKeyboardEvent(e);
+				if (e.repeat) {
+					e.stopImmediatePropagation();
```

Comment:
Is there a reason you're not stopping propagation for any of the events we handle? Because this is now going to block any keyboard events from being propagated up if they're held down which probably isn't what we want. 

I think it would just be simplest to keep track of whether we handled the event (boolean that's set in each of the if checks) and if so then stop propagation.